File: hadoop_201809/hdfs-demo/src/main/java/mr/MyPartitioner.java
package mr;

import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Partitioner;

/**
 * @Description:
 * @author: HuangYn
 * @date: 2018/9/4 21:02
 */
public class MyPartitioner extends Partitioner<Text, IntWritable> {

    @Override
    public int getPartition(Text text, IntWritable intWritable, int numPartitions) {

        System.out.println("=============================");

        return 0;
    }
}

File: spark/spark-demo/src/main/scala/chapter4/MyPartitioner.scala
package chapter4

import org.apache.spark.Partitioner

/**
  * @Description:
  * @author: HuangYn 
  * @date: 2018/10/22 19:13
  */
class MyPartitioner(numParts: Int) extends Partitioner {

  override def numPartitions: Int = numParts

  override def getPartition(key: Any): Int = {
    if (key.toString.toInt % 2 == 0) {
      0
    } else 1
  }

  override def equals(ohter: scala.Any): Boolean = {
    ohter match {
      case mypartition: MyPartitioner => mypartition.numPartitions == numPartitions
      case _ => false
    }
  }
}
